core/toolbox/category.js

```diff
@@ -0,0 +1,269 @@
+   * The name that will be displayed on the category.
+   * @type {string}
+   * @protected
+   */
+  this.name_ = categoryDef['name'];
+
+  var contents = categoryDef['contents'];
+
+  /**
+   * True if this category has subcategories, false otherwise.
+   * @type {boolean}
+   * @private
+   */
+  this.hasChildren_ = contents && contents.length &&
+    typeof contents != 'string' &&
+    contents[0].kind.toUpperCase() == 'CATEGORY';
+  /**
+   * The parent of the category.
+   * @type {?Blockly.ToolboxCategory}
+   * @protected
+   */
+  this.parent_ = opt_parent || null;
+
+  /**
+   * The level that the category is nested at.
+   * @type {number}
+   * @protected
+   */
+  this.level_ = this.parent_ ? this.parent_.getLevel() + 1 : 0;
+
+  /**
+   * The colour of the category.
+   * @type {string}
+   * @protected
+   */
+  this.colour_ = this.getColour_(categoryDef);
+
+  /**
+   * The html container for the category.
+   * @type {?Element}
+   * @protected
+   */
+  this.htmlDiv_ = null;
+
+  /**
+   * The html element for the category row.
+   * @type {?Element}
+   * @protected
+   */
+  this.rowDiv_ = null;
+
+  /**
+   * The html element for the toolbox icon.
+   * @type {?Element}
+   * @protected
+   */
+  this.iconSpan_ = null;
+
+  /**
+   * Container for any children categories.
+   * @type {?Element}
+   * @protected
+   */
+  this.subcategoriesDiv_ = null;
+
+  /**
+   * The config for all the category css classes.
+   * @type {!Blockly.ToolboxCategory.ClassConfig}
+   * @protected
+   */
+  this.classConfig_ = {
+    'container': 'blocklyToolboxCategory',
+    'row': 'blocklyTreeRow',
+    'icon': 'blocklyTreeIcon',
+    'label': 'blocklyTreeLabel',
+    'contents': 'blocklyToolboxContents',
+    'selected': 'blocklyTreeSelected',
+    'openIcon': 'blocklyTreeIconOpen',
+    'closedIcon': 'blocklyTreeIconClosed'
+  };
+
+  Blockly.utils.object.mixin(this.classConfig_, categoryDef['classConfig']);
+
+  /**
+   * Whether or not the category should display it's children.
+   * @type {boolean}
+   * @protected
+   */
+  this.expanded_ = categoryDef['expanded'] == 'true' || categoryDef['expanded'];
+
+  /**
+   * True if the category is visible, false otherwise.
+   * @type {boolean}
+   * @private
+   */
+  this.isVisible_ = true;
+
+  /**
+   * Parse the contents for this category.
+   * @type {string|
+   *        !Array<!Blockly.ToolboxItem>|
+   *        !Array<!Blockly.utils.toolbox.FlyoutItemDef>}
+   * @protected
+   */
+  this.contents_ = this.parseContents_(categoryDef, this.hasChildren_);
+};
+
+Blockly.utils.object.inherits(Blockly.ToolboxCategory,
+    Blockly.CollapsibleToolboxItem);
+
+/**
+ * @typedef {{
+ *            container:?string,
+ *            row:?string,
+ *            icon:?string,
+ *            label:?string,
+ *            contents:?string,
+ *            selected:?string,
+ *            openIcon:?string,
+ *            closedIcon:?string,
+ *          }}
+ */
+Blockly.ToolboxCategory.ClassConfig;
```

Comment:
Yeah that is a good question, I think I was saving core/utils/toolbox for typedefs that are used throughout the code base or for typedefs that are ambiguous whether they belong on the toolbox or the flyout. In this case since the class config is very specifically for the Category I thought I would leave it here.